src/main.go

```
@@ -0,0 +1,195 @@
+		router := mux.NewRouter()
+		store := sessions.NewCookieStore([]byte("109ae017-5e40-476a-bdba-d77b62a8e377"))
+
+		// create controller dependencies
+		kubernetesManager, err := auth.NewKubernetesManager(
+			c.Bool(config.FlagInCluster),
+			c.String(config.FlagKubeConfigPath),
+			nil,
+		)
+		if err != nil {
+			return err
+		}
+
+		auth0Authenticator := auth.NewAuth0Authenticator(
+			c.String(config.FlagAuth0Domain),
+			c.String(config.FlagAuth0Connection),
+			c.String(config.FlagAuth0ClientID),
+			c.String(config.FlagClusterName),
+			newHttpClient(),
+		)
+		awsManager := auth.NewAWSManager(
+			c.String(config.FlagClusterName),
+			c.String(config.FlagAWSRegion),
+		)
+
+		// create controllers
+		cs := []interface {
+			GetRoutes() []controllers.Route
+		}{
+			controllers.NewAuthController(auth0Authenticator, store),
+			controllers.NewAWSController(awsManager, kubernetesManager, store),
+		}
+
+		// register routes for all the controllers
+		restrictedRouteMap := map[string]bool{}
+		for _, controller := range cs {
+			for _, route := range controller.GetRoutes() {
+				router.HandleFunc(route.Path, route.Handler).Methods(route.Method...)
+				restrictedRouteMap[route.Path] = route.Restricted
+			}
+		}
+
+		// register file server at root
+		fs := http.FileServer(http.Dir("./ui/build"))
+		router.PathPrefix("/").Handler(http.StripPrefix("/", fs))
+
+		// add middleware
+		router.Use(controllers.NewLoggingMiddleware())
+		router.Use(controllers.NewAuthenticationMiddleware(restrictedRouteMap, store))
+		router.Use(controllers.NewCredentialsMiddleware(restrictedRouteMap, store))
+
+		// create server and start listening
+		srv := &http.Server{
+			Handler:      router,
+			Addr:         "0.0.0.0:" + c.String("port"),
+			WriteTimeout: 15 * time.Second,
+			ReadTimeout:  15 * time.Second,
+		}
+
+		go func() {
+			log.Println("[INFO] listening on", srv.Addr)
+			if err := srv.ListenAndServe(); err != nil {
+				log.Println("[ERROR]", err)
```

this should probably be `log.Fatal` so we fail fast

---

✅